src/backend/api/Fusion.Resources.Domain/Queries/GetRelevantOrgUnits.cs

```diff
@@ -0,0 +1,332 @@
+                {
+                    FullDepartment = dep,
+                    Reason = "DelegatedManager"
+                }));
+                //retList.AddRange(relevantSectors.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep,
+                //    Reason = "RelevantSector"
+                //}));
+                //retList.AddRange(relevantDepartments.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep.DepartmentId,
+                //    Reason = "RelevantDepartment"
+                //}));
+                if (isDepartmentManager) orgUnitAccessReason.AddRange(lineOrgDepartmentProfile?.Children.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep.DepartmentId,
+                    Reason = "ParentManager"
+                }) ?? Array.Empty<QueryOrgUnitReason>());
+                //retList.AddRange(lineOrgDepartmentProfile?.Siblings.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep.DepartmentId,
+                //    Reason = "RelevantSibling"
+                //}) ?? Array.Empty<QueryOrgUnit>());
+                orgUnitAccessReason.AddRange(delegatedParentManagerWithResposibility?.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep.DepartmentId,
+                    Reason = "DelegatedParentManager"
+                }) ?? Array.Empty<QueryOrgUnitReason>());
+
+                List<QueryRelevantOrgUnit> populatedOrgUnitResult = GetRelevantOrgUnits(orgUnits, orgUnitAccessReason);
+
+                var filteredOrgUnits = ApplyOdataFilters(request.Query, populatedOrgUnitResult.OrderBy(x => x.SapId));
+
+                var skip = request.Query.Skip.GetValueOrDefault(0);
+                var take = request.Query.Top.GetValueOrDefault(100);
+
+                filteredOrgUnits.DistinctBy(x => x.SapId);
+
+                var pagedQuery = QueryRangedList.FromEnumrableItems(filteredOrgUnits, skip, take);
+
+                return pagedQuery;
+            }
+
+            private static List<QueryRelevantOrgUnit> GetRelevantOrgUnits(IEnumerable<QueryRelevantOrgUnit> cachedOrgUnits, List<QueryOrgUnitReason> orgUnitAccessReason)
+            {
+                var endResult = new List<QueryRelevantOrgUnit>();
+                foreach (var org in orgUnitAccessReason)
+                {
+                    if (org?.FullDepartment != null && org?.Reason != null)
+                    {
+                        var alreadyInList = endResult.FirstOrDefault(x => x.FullDepartment == org.FullDepartment);
+
+                        if (alreadyInList is null)
+                        {
+                            var data = cachedOrgUnits.Where(x => x.FullDepartment == org?.FullDepartment.Replace('*', ' ').TrimEnd()).FirstOrDefault();
```

Comment:
if you really want to do this, you can replace .Where with .FirstOrDefault (and remove .FirstOrDefault from enf of line)